datafusion/datasource/src/schema_adapter.rs

```diff
@@ -98,6 +100,46 @@ pub trait SchemaMapper: Debug + Send + Sync {
     fn map_batch(&self, batch: RecordBatch) -> datafusion_common::Result<RecordBatch>;
 }
 
+pub trait MissingColumnGeneratorFactory: Debug + Send + Sync {
```

Comment:
I haven't had a chance to review this entire PR yet, but I am feeling a little strange that these traits are not used by `SchemaMapper` but instead are used by `DefaultSchemaMapper`

I wonder what is the usecase for these extra two traits, compared to say adding a method on `SchemaMapper`